### ISSUE TITLE
Style/RedundantLineContinuation should not look inside strings

### ISSUE THREAD

**Issue by bquorning** [rubocop/rubocop#11758]
When the trailing backslash is inside a string literal, I think it needs to be disregarded by the Style/RedundantLineContinuation cop.

--------

## Expected behavior

I expect these two spec examples to pass in spec/rubocop/cop/style/redundant_line_continuation_spec.rb:

```rb
expect_no_offenses(<<~'RUBY')
  "foo \
    bar"
RUBY

expect_no_offenses(<<~'RUBY')
  'foo \
    bar'
RUBY
```

## Actual behavior

The spec examples fail, suggesting to remove the trailing backslash.

## Steps to reproduce the problem

See above.

## RuboCop version

7c35167233e567cd0bad3879974de787b3c372bd

**Comment by bquorning** [comment]
cc @koic, @ydah

**Comment by koic** [comment]
@bquorning Yeah, I've opened https://github.com/rubocop/rubocop/pull/11759 to fix the issue.
